src/AudioAnalysisTools/Events/EventFilters.cs

```diff
@@ -149,41 +213,50 @@ public static List<EventCommon> FilterOnDuration(List<EventCommon> events, doubl
         /// </summary>
         public static List<EventCommon> FilterEventsOnSyllableCountAndPeriodicity(List<EventCommon> events, int maxSyllableCount, double expectedPeriod, double expectedSd)
         {
-            var minExpectedPeriod = expectedPeriod - (3 * expectedSd);
-            var maxExpectedPeriod = expectedPeriod + (3 * expectedSd);
+            var SigmaThreshold = 3.0;
+            var minExpectedPeriod = expectedPeriod - (SigmaThreshold * expectedSd);
+            var maxExpectedPeriod = expectedPeriod + (SigmaThreshold * expectedSd);
 
             var filteredEvents = new List<EventCommon>();
 
+            int count = 0;
             foreach (var ev in events)
             {
+                count++;
+
                 // ignore non-composite events
                 if (ev is CompositeEvent == false)
                 {
                     filteredEvents.Add(ev);
+                    Log.Debug($" Event{count} accepted one syllable.");
                     continue;
                 }
 
                 // Get the temporal footprint of the component events.
                 (bool[] temporalFootprint, double timeScale) = GetTemporalFootprint(ev);
 
-                // calculate the periodicity in seconds
+                // calculate the actual periods in seconds
                 int syllableCount = 1;
-                var periodSeconds = new List<double>();
+                var actualPeriodSeconds = new List<double>();
                 int previousEventStart = 0;
                 for (int f = 1; f < temporalFootprint.Length; f++)
                 {
                     if (temporalFootprint[f] && !temporalFootprint[f - 1])
                     {
                         // calculate the event interval in seconds.
                         syllableCount++;
-                        periodSeconds.Add((f - previousEventStart + 1) * timeScale);
+                        actualPeriodSeconds.Add((f - previousEventStart + 1) * timeScale);
                         previousEventStart = f;
                     }
                 }
 
-                // reject composite events whose total syllable count exceeds the user defined max.
+                string strArray = DataTools.Array2String(actualPeriodSeconds.ToArray());
+                Log.Debug($" Event{count} actual periods: {strArray}");
```

Comment:
Use https://github.com/QutEcoacoustics/audio-analysis/blob/5baf2d932232da4c72f6972e921736039dfc6474/src/Acoustics.Shared/Extensions/EnumerableExtensions.cs#L307-L319

```suggestion
                Log.Debug($" Event{count} actual periods: {actualPeriodSeconds.Join(",")}");
```

Reply:
New extension method has been written.